awd/cli.py

```diff
@@ -1,7 +1,10 @@
+from __future__ import annotations
```

Comment:
I noticed this was added to a handful of files, just curious, why is that?  Was it linting error introduced by `ruff`?

I get the impression there is some interesting, but complex, decisions made with [PEP 563](https://peps.python.org/pep-0563/) and [PEP 649](https://peps.python.org/pep-0649/).  My understanding, just reading today, is that PEP 563 -- using this import convention -- turned type annotations into strings and therefore did not require imports.  While PEP 649 makes this default behavior, and therefore does _not_ require this import.

I see from the `Pipfile` and `Dockerfile` that this is a Python 3.11 project so we might not need this, even if a linter is throwing errors.  It certainly doesn't seem to hurt by importing it, but it might be worth considering that we don't require it for projects 3.11+.

Reply:
Ruff catches errors with these import statements through rule [[UP010] unnecessary-future-import](https://beta.ruff.rs/docs/rules/unnecessary-future-import/#unnecessary-future-import-up010).

Thought I'd share that  when I applied linting with Ruff for a Python 3.11 project (`carbon`),  it threw an error `from __future__ import absolute_import` but not the statement above. 🤔

Reply:
Thanks to @jonavellecuerdo for pushing on this a bit in a slack conversation, I may be wrong about PEP 563 and 649, or at least there is more nuance.

Some of that nuance may be around:
```python
from mypy_boto3_dynamodb.type_defs import PutItemOutputTypeDef
```

If `PutItemOutputTypeDef` is used as a type in `dynamodb.py`, but `mypy` is only a Pipenv `--dev` install, then `mypy_boto3_dynamodb` won't be available in the deployed environment.  Therefore, this seems to be required:

```python
if TYPE_CHECKING:
    from mypy_boto3_dynamodb.type_defs import PutItemOutputTypeDef
```

But for _that_ to work, then you **do** need `from __future__ import annotations` at the top, otherwise it throws an error during `make test` (and theoretically when deployed as well):

```
def add_doi_item_to_database(self, doi_table: str, doi: str) -> PutItemOutputTypeDef:
E   NameError: name 'PutItemOutputTypeDef' is not defined
```

This would seems to suggest if we **do** want to use `mypy_boto3_dynamodb` as annotated types, but we **don't** want to have `from __future__ import annotations` at the top, then we'd need to quote the types in the function definitions, e.g.:

```python
    def add_doi_item_to_database(
        self, doi_table: str, doi: str
    ) -> "PutItemOutputTypeDef":  #<------------------
```

Given that it is currently in place, and in multiple files, I'm certainly okay with leaving `from __future__ import annotations` as it appears to work and the type annotation is more than just a string (which is nice for IDE's to see the definition).